src/Transport/Seam/MessagePump.cs

```diff
@@ -56,10 +56,10 @@ public Task Init(Func<PushContext, Task> pump, CriticalError criticalError, Push
         // For internal testing purposes.
         internal void OnError(Func<Exception, Task> func)
         {
-            topologyOperator.OnError(exception =>
+            topologyOperator.OnError(async exception =>
```

Comment:
The change failed the test and I've found an issue with CB thanks to that. Great :)